nservicebus/transactional-session/index.md

```diff
@@ -140,19 +169,31 @@ Internally the transactional session doesn't use a single transaction that spans
 9. The transaction is committed, and the outbox record and business data modifications are stored atomically.
 
 ### Phase 2
 
 The endpoint receives the control message and processes it as follows:
-   * Find the outbox record.
-   * If it exists, it hasn't been marked as dispatched, and there are pending operations, dispatch those operations, and set the outbox record as dispatched.
-   * If it doesn't exist, processing of the control message is delayed to a later point.
 
-## Important design consideration
+* Find the outbox record.
+  * If it exists, it hasn't been marked as dispatched, and there are pending operations they are dispatched, and the outbox record is set as dispatched.
+  * If it doesn't exist, delay the processing of the control message.
+
+## Failure scenarios
+
+The transactional session provides atomic store-and-send guarantees, similar to Outbox (except for incoming message de-duplication). That said, it cannot rely on the recoverability mechanism used by the Outbox, which uses retries to ensure outgoing messages are dispatched when failures occur. Instead, the control message is used to ensure that **exactly one** of these two outcomes occur:
+
+* Transaction finishes with no visible side effects - when the control message stores the `OutboxRecord`
+* Transaction finishes with data being stored, and outgoing messages eventually sent - when the `Commit` path successfully stores the `OutboxRecord`
+
+Sending the control message first ensures that, eventually, the transaction will have an atomic outcome. If the `Commit` of the `OutboxRecord` succeeds, the control message will ensure the outgoing operations are sent out. If the `Commit` fails, the control message will (after the [maximum commit duration](#usage-advanced-maximum-commit-duration) elapses) eventually be consumed, leaving no side effects.
+
+When dispatching the control message fails, the transactional session changes will be rolled back, and an error will be raised to the user committing the session.
+
+## Limitations
 
-* The transactional session uses a control message that is sent to the local queue. Due to this design, this feature requires a full endpoint and cannot be used in send-only endpoints.
-* Deduplication is guaranteed in Phase 2, but not in Phase 1. In Phase 2, the outbox record ensures that the operations will never be dispatched more than once. However, during Phase 1, a unique ID is assigned to the session. At that point, the user is responsible for ensuring that no duplicate requests are executed.
+* The transactional session cannot be used in send-only endpoints. A full endpoint is required to send a control message to the local queue.
+* Deduplication is guaranteed in Phase 2, but not in Phase 1. In Phase 2, the Outbox record ensures that the operations will never be dispatched more than once. However, in Phase 1, the session is always assigned a unique identifier and thus the operations in Phase 1 cannot be retried.
```

Comment:
the outbox mechanism only marks all dispatched messages as dispatched after all have been sent. When dispatching 3 messages and the third one fails, the process will attempt to re-dispatch all 3 messages again, doesn't it?

Reply:
Removed the whole bullet after discussing it with @tmasternak and @SzymonPobiega since we agreed it doesn't convey much anyway